common/src/main/java/earth/terrarium/hermes/api/defaults/FillAndBorderElement.java

```diff
@@ -0,0 +1,81 @@
+package earth.terrarium.hermes.api.defaults;
+
+import com.teamresourceful.resourcefullib.common.color.Color;
+import com.teamresourceful.resourcefullib.common.color.ConstantColors;
+import earth.terrarium.hermes.api.TagElement;
+import earth.terrarium.hermes.utils.ElementParsingUtils;
+import net.minecraft.client.gui.GuiGraphics;
+
+import java.util.Map;
+import java.util.function.IntFunction;
+
+public abstract class FillAndBorderElement implements TagElement {
+
+    /* For background fills and borders.
+    Though terms are borrowed from CSS's Box Model, does not change Hermes' layout model.
+    Does allow inheriting elements to be set taller and/or wider.
+    The area behind the element, plus any padding, may be filled with background color
+    A border may surround the padding area, with its own width and color. */
+
+    protected int backgroundPadding = 0;
+    protected Color backgroundColor = ConstantColors.gray;
+    protected boolean hasBackground = false;
+    protected int borderWidth = 0;
+    protected Color borderColor = ConstantColors.whitesmoke;
+    protected boolean hasBorder = false;
+    protected int hSpacing; // (background + borderWidth);
```

Comment:
spacing is for between elements not edges these should be called xPadding and yPadding

Reply:
That name would be confusing with the background padding.
Perhaps `xMargin` and `yMargin` would be better? Also fits with the terminology from CSS (not that we're implementing CSS, but seemed like a decent external convention to reference for names for what's going on here.

![image](https://github.com/terrarium-earth/Hermes/assets/6677700/82699bff-0295-45ae-b6e5-6b51e02b057f)

Reply:
Renamed [h|v]Spacing to [x|y]Margin with 8826712.

Reply:
Hrm.

On second thought, 'margin' isn't ideal, as we are trying to name variables that represents the sum of "padding" and "border". Doesn't conflict with any other current variable names in the Hermes thought. I'll see if I can think of a still better name. <chin stroking intensifies...>

Reply:
I personally think padding is fine as backgroundPadding is that a background padding either way the background padding should be equal to padding full padding as to go inline with css borders are on the outside of content while padding is inwards. Margin wouldnt be correct as it is outwards.

Reply:
OK, that's a difference from CSS; in this patch `backgroundPadding` extends outward frame the base/unadorned area the element takes up. Like in the last sample photo in the opening post of this PR... the red fill area extends past the strict region of the text; the `borderWidth` is then drawn outside that.

*How about "xSurround" and "ySurround"?*

In use, they are offsets from x,y for the actual element rendering, and come up in getHeight() and the width passed to font.split(). When fill or border will be drawn, they are the sum of those; when not, they become the same offsets that were in Hermes before, (0 for horizontal, but 1 above and below).

Reply:
5c62755 renames [x|y]Margin to [x|y]Surround.